rpc.go

```diff
@@ -328,10 +358,10 @@ func (t *Topic) resMessageHandler(from peer.ID, topic string, msg []byte) ([]byt
 
 	log.Debugf("%s response from %s: %s", topic, from, res.ID)
 	t.lk.Lock()
-	ch := t.resChs[id]
+	m, exists := t.ongoing[id]
 	t.lk.Unlock()
-	if ch != nil {
-		ch <- res
+	if exists && m.respCh != nil {
+		m.respCh <- res
 	} else {
 		log.Warnf("%s missed response from %s: %s", topic, from, res.ID)
```

Comment:
I removed this line in the other PR since it's always called when `args.ignoreResponse` is true. It could also be moved to a debug message (either way). Warn doesn't sound right for expected behavior.

Another option: with your changes the `ongoingMessage` should always exist, meaning we could log a warning if it doesn't:

```
if exists {
	if m.respCh != nil {
		m.respCh <- res
	}
} else {
	log.Warnf("%s missed response from %s: %s", topic, from, res.ID)
}
```

Reply:
Will do

Reply:
missing from `ongoingMessage` actually means the response arrives too late - either the context deadline is hit, or other responses cause the goroutine to exit. It can happen, so I'll change it to debugf as well.